msg/versioned/ArmingCheckReply.msg

```diff
@@ -1,34 +1,38 @@
+# Arming check reply.
+#
+# Response to an ArmingCheckRequest message.
+
 uint32 MESSAGE_VERSION = 0
 
-uint64 timestamp # time since system start (microseconds)
+uint64 timestamp # Time since system start [us]
 
-uint8 request_id
-uint8 registration_id
+uint8 request_id # Id of ArmingCheckRequest for which this is a response.
+uint8 registration_id # Id of component emitting this response.
 
-uint8 HEALTH_COMPONENT_INDEX_NONE = 0
+uint8 HEALTH_COMPONENT_INDEX_NONE = 0 # Index of health component for which this response applies.
 
-uint8 health_component_index      # HEALTH_COMPONENT_INDEX_*
-bool health_component_is_present
-bool health_component_warning
-bool health_component_error
+uint8 health_component_index # HEALTH_COMPONENT_INDEX_*
+bool health_component_is_present ?
+bool health_component_warning # ?
+bool health_component_error # ?
 
-bool can_arm_and_run              # whether arming is possible, and if it's a navigation mode, if it can run
+bool can_arm_and_run # True if arming is possible, and for a navigation mode, if it can run.
 
-uint8 num_events
+uint8 num_events # ?
 
-Event[5] events
+Event[5] events # ?
 
 # Mode requirements
-bool mode_req_angular_velocity
-bool mode_req_attitude
-bool mode_req_local_alt
-bool mode_req_local_position
-bool mode_req_local_position_relaxed
-bool mode_req_global_position
-bool mode_req_mission
-bool mode_req_home_position
-bool mode_req_prevent_arming
-bool mode_req_manual_control
-
-
-uint8 ORB_QUEUE_LENGTH = 4
+bool mode_req_angular_velocity # ?
```

Comment:
How are these all used? My guess would be that a registration is code representing a mode. If a mode gets the arming check request, it responds with a replay indicating its requirements, which will be some set of these.
PX4 "elsewhere" would check if these requirements are met - is that right?

Presumably all modes have to report their arming check - does that mean every registration (mode) will have its own message instance that it will need to write to - and so perhaps it doesn't have to update these requirements each time.

I'm sure you can see I am confused by the architecture!

Reply:
> is that right?

Yes.
My first reply should make it clearer.